pennylane/devices/default_mixed.py

```diff
@@ -377,3 +377,56 @@ def preprocess(
         )
 
         return transform_program, config
+
+
+# The following part is for recording some legacy code that might be useful for future reference.
+def _apply_state_vector(total_wires, state, device_wires, interface="numpy", tolerance=1e-10):
+    """Initialize the internal state in a specified pure state.
+
+    Args:
+        total_wires (Wires): Total wires in the system
+        state (array[complex]): normalized input state of length
+            ``2**len(device_wires)``
+        device_wires (Wires): wires that get initialized in the state
+        interface (str): Interface to use for array operations. Defaults to "numpy".
+        tolerance (float): Tolerance for checking if state is normalized. Defaults to 1e-10.
+
+    Returns:
+        array[complex]: The density matrix for the full system after applying the state
+    """
+    num_total_wires = len(total_wires)
+
+    state = qml.math.asarray(state, like=interface)
+    n_state_vector = state.shape[0]
+
+    if state.ndim != 1 or n_state_vector != 2 ** len(device_wires):
+        raise ValueError("State vector must be of length 2**wires.")  # pragma: no cover
```

Comment:
What if the state is batched?

Reply:
This is confusing due to need of test coverage of `qml.math.scatter`. Scatter used to be only called by legacy mixed device; however we also do not want to delete it directly since this kind of helper methods could be useful for future dev. They won't take too much resources loading; so I just remain the legacy usage here to test it.
Already refactored the test structure such that this piece of code only shows up in `tests/math/test_fucntions`